lib/trino-parquet/src/test/java/io/trino/parquet/TestTupleDomainParquetPredicate.java

```diff
@@ -415,6 +418,89 @@ public void testTimestamp(int precision, LocalDateTime baseTime, Object baseDoma
+                {LogicalTypeAnnotation.TimeUnit.NANOS, 9, baseTime, nanosExpectedValue},
+        };
+    }
+
+    private LocalDateTime baseTimeWithNanoOfSecond(int nanoOfSecond)
+    {
+        return LocalDateTime.of(1970, 1, 19, 10, 28, 52, nanoOfSecond);
+    }
+
+    @Test(dataProvider = "int64TimestampPrecision")
+    public void testTimestampInt64(LogicalTypeAnnotation.TimeUnit timeUnit, int precision, LocalDateTime baseTime, Object baseDomainValue)
+            throws ParquetCorruptionException
+    {
+        int parquetPrecision;
+        switch (timeUnit) {
+            case MILLIS:
+                parquetPrecision = 3;
+                break;
+            case MICROS:
+                parquetPrecision = 6;
+                break;
+            case NANOS:
+                parquetPrecision = 9;
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown Parquet TimeUnit " + timeUnit);
+        }
+
+        PrimitiveType type = Types.required(INT64)
+                .as(LogicalTypeAnnotation.timestampType(true, timeUnit))
+                .named("TimestampColumn");
+
+        ColumnDescriptor columnDescriptor = new ColumnDescriptor(new String[]{}, type, 0, 0);
+        TimestampType timestampType = createTimestampType(precision);
+        assertEquals(getDomain(columnDescriptor, timestampType, 0, null, ID, UTC), all(timestampType));
+        LocalDateTime maxTime = baseTime.plus(Duration.ofMillis(50));
+
+        Object maxDomainValue;
+        if (baseDomainValue instanceof Long) {
+            maxDomainValue = (long) baseDomainValue + 50 * MICROSECONDS_PER_MILLISECOND;
+        }
+        else if (baseDomainValue instanceof LongTimestamp) {
+            LongTimestamp longTimestamp = ((LongTimestamp) baseDomainValue);
+            maxDomainValue = new LongTimestamp(longTimestamp.getEpochMicros() + 50 * MICROSECONDS_PER_MILLISECOND, longTimestamp.getPicosOfMicro());
+        }
+        else {
+            throw new IllegalArgumentException("Unknown Timestamp domain type " + baseDomainValue);
+        }
+
+        long minValue = toEpochWithPrecision(baseTime, parquetPrecision);
+        long maxValue = toEpochWithPrecision(maxTime, parquetPrecision);
+        assertEquals(getDomain(columnDescriptor, timestampType, 10, longColumnStats(minValue, minValue), ID, UTC), singleValue(timestampType, baseDomainValue));
+        assertEquals(getDomain(columnDescriptor, timestampType, 10, longColumnStats(minValue, maxValue), ID, UTC),
+                Domain.create(ValueSet.ofRanges(range(timestampType, baseDomainValue, true, maxDomainValue, true)), false));
+    }
+
+    private static long toEpochWithPrecision(LocalDateTime time, int precision)
+    {
+        long scaledEpochSeconds = time.toEpochSecond(ZoneOffset.UTC) * (long) Math.pow(10, precision);
+        long fractionOfSecond = Math.round(1.0 * time.getNano() / (long) Math.pow(10, 9 - precision));
```

Comment:
`LongMath.divide( .. . ROUNDING_UNNECESSARY)`, so that we don't need to define method behavir with respect to rounding/truncation.